File: hostinfo.py
#!/usr/bin/env python

import os
import flask

app = flask.Flask('hostinfo')

@app.route('/')
def hostinfo():
    result = {
        'env': dict(os.environ),
        'headers': dict(flask.request.headers),
        'request': {
            'client': flask.request.remote_addr,
            'host': flask.request.host,
            'scheme': flask.request.scheme
        }
    }
    return flask.jsonify(result)

if __name__ == '__main__':
    app.run(host='0.0.0.0')

File: hostinfo.js
#!/usr/bin/env node

const { env } = require("process");

const express = require("express");

const port = env.HOSTINFO_PORT || 3000;

const app = express();
app.set("json spaces", 2);

app.use((req, res) =>
  res.send({
    env,
    headers: req.headers,
    request: {
      method: req.method,
      client: req.ip,
      host: req.hostname,
      scheme: req.protocol
    }
  })
);

app.listen(port, () => console.log(`Server is listening on port ${port}`));
